### DISCUSSION TITLE
Alternative workflow topology to avoid "workflow reference count exceeds the limit of 20" error

### DISCUSSION THREAD

**Discussion by ckelleyRH** [community/community#32918] [Actions]
**Background context**
In my [repo](https://github.com/dogtagpki/pki/pull/4153/checks) I have 12 workflows that begin with a job to determine the version of Fedora to test against and then a job to build the application/various docker images. This results in 12 parallel jobs, 11 of which are redundant, each of which takes ~10 minutes. I want to remove these wasteful runs.

**What I tried already**

I created a reusable workflow for the common jobs. I then created a "main workflow" to call the 12 workflows, reusing the artefacts from the reusable workflow. This doesn't work as the component jobs in each workflow are run from other workflow files and there is a limit of 20 workflows that can be called from 1 workflow. This limit applies globally; i.e. I made 1 workflow, calling 2 workflows, each calling 10 workflows, as a test and it still failed. Incidentally, I can't find this documented anywhere. If it is documented somewhere please help me find it!

I considered having the common reusable workflow call the other workflows on completion, but as far as I am aware it is not possible to pass an artefact from one workflow to another on completion. There is a REST endpoint to download artefacts but I don't know how this plays with PRs, whether all the subsequently triggered workflows will be associated with the original PR for example. I figured there must be a cleaner way.

**Does anyone have a similar problem? How did you get around it?**

**Comment by MatanHeledPort** [comment]
I just encountered the same issue.

This also seems to happen even if you dont really call the workflows, and there are if statesments between them

Any solution or update ?


`
